File: library/src/main/java/com/yarolegovich/slidingrootnav/transform/CompositeTransformation.java
package com.yarolegovich.slidingrootnav.transform;

import android.view.View;

import java.util.List;

/**
 * Created by yarolegovich on 25.03.2017.
 */

public class CompositeTransformation implements RootTransformation {

    private List<RootTransformation> transformations;

    public CompositeTransformation(List<RootTransformation> transformations) {
        this.transformations = transformations;
    }

    @Override
    public void transform(float dragProgress, View rootView) {
        for (RootTransformation t : transformations) {
            t.transform(dragProgress, rootView);
        }
    }
}

File: library/src/main/java/com/yarolegovich/slidingrootnav/transform/CompositeTransformation.kt
package com.yarolegovich.slidingrootnav.transform

import android.view.View

/**
 * Created by yarolegovich on 25.03.2017.
 * Edited by Mehdi on 27.02.2019
 */

class CompositeTransformation(private val transformations: List<RootTransformation>)
    : RootTransformation {

    override fun transform(dragProgress: Float, rootView: View) {
        for (t in transformations) {
            t.transform(dragProgress, rootView)
        }
    }
}
